File: Infinity.Bnois.Web/Infinity.Bnois.Web/app/services/currentStatusService.js
(function () {
    'use strict';
    angular.module('app').service('currentStatusService', ['dataConstants', 'apiHttpService', currentStatusService]);

    function currentStatusService(dataConstants, apiHttpService) {
        var service = {
            getGeneralInformation: getGeneralInformation,
            getCivilAcademicQualification: getCivilAcademicQualification,
            getSecurityClearance: getSecurityClearance,
            getCourseAttended: getCourseAttended,
            getForeignCourseAttended: getForeignCourseAttended,
            getExamTestResult: getExamTestResult,
            getCareerForecast: getCareerForecast,
            getCarLoanInfo: getCarLoanInfo,
            getPunishmentDiscipline: getPunishmentDiscipline,
            getCommendationAppreciation: getCommendationAppreciation,
            getAward: getAward,
            getMedal: getMedal,
            getCleanService: getCleanService,
            getPublication: getPublication,
            getChildren: getChildren,
            getSibling: getSibling,
            getNextOfKin: getNextOfKin,
            getHeir: getHeir,
            getOprGrading: getOprGrading,
            getForeignVisit: getForeignVisit,
            getForeignCourseVisitGrandTotal: getForeignCourseVisitGrandTotal,
            getParentInfo: getParentInfo,
            getSpouseInfo: getSpouseInfo,
            getTransferHistory: getTransferHistory,
            getCostGuardHistory: getCostGuardHistory,
            getPromotionHistory: getPromotionHistory,
            getAdditionalSeaServices: getAdditionalSeaServices,
            getSeaServices: getSeaServices,
            getInstructionalServices: getInstructionalServices,
            getSeaCommandServices: getSeaCommandServices,
            getInterOrganizationServices: getInterOrganizationServices,
            getIntelligenceServices: getIntelligenceServices,
            getCurrentStatus: getCurrentStatus,
            getNotifications: getNotifications,
            getRemark: getRemark,
            getPersuasion: getPersuasion,
            getCourseFuturePlan: getCourseFuturePlan,
            getTransferFuturePlan: getTransferFuturePlan,
            getTemporaryTransferHistory: getTemporaryTransferHistory,
            getLeaveInfo: getLeaveInfo,
            getAdminAuthorityService: getAdminAuthorityService,
            getISSB: getISSB,
            getForeignProjects: getForeignProjects,
            getMissions: getMissions,
            getHODServices: getHODServices,
            getDockyardServices: getDockyardServices,
            getShoreCommandServices: getShoreCommandServices,
            getSubmarineServices: getSubmarineServices,
            getDeputationServices: getDeputationServices,
            getOutsideServices: getOutsideServices,
            GetFamilyPermissionRelationCount: GetFamilyPermissionRelationCount,
            GetFamilyPermissions: GetFamilyPermissions,
            getZoneServices: getZoneServices,
            getUnmDeferment : getUnmDeferment,
            getZoneCourseMissionServices: getZoneCourseMissionServices,
            getMscEducationQualification: getMscEducationQualification,
            getBatchPosition: getBatchPosition
           
        };

        return service;
     
        function getGeneralInformation(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-general-information?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getCivilAcademicQualification(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-civil-academic-qualification?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getMscEducationQualification(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-msc-education-qualification?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getSecurityClearance(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-security-clearance?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getCourseAttended(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-course-attended?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getUnmDeferment(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-unm-deferment?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getForeignCourseAttended(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-foreign-course-attended?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getExamTestResult(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-exam-test-result?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getCareerForecast(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-career-forecast?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getCarLoanInfo(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-car-loan-info?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        

        function getBatchPosition(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-batch-position?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getPunishmentDiscipline(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-punishment-discipline?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getCommendationAppreciation(pNo,type) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-commendation-appreciation?pNo=' + pNo+'&type='+type;
            return apiHttpService.GET(url);
        }
        function getAward(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-award?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getMedal(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-medal?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getPublication(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-publication?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getCleanService(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-clean-service?pNo=' + pNo;
            return apiHttpService.GET(url);
        }


        function getChildren(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-current-children?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getSibling(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-current-sibling?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getNextOfKin(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-current-next-of-kin?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getHeir(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-heir-info?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getOprGrading(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-opr-grading?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getForeignVisit(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-foreign-visit?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getForeignCourseVisitGrandTotal(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-foreign-course-visit-grand-total?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getParentInfo(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-parent-info?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getSpouseInfo(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-spouse-info?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getTransferHistory(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-transfer-history?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getCostGuardHistory(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-costguard-history?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getTemporaryTransferHistory(transferId) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-temporary-transfer-history?transferId=' + transferId;
            return apiHttpService.GET(url);
        }

        function getPromotionHistory(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-promotion-history?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getAdditionalSeaServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-additional-sea-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getSeaServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-sea-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }


        function getZoneServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-zone-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getZoneCourseMissionServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-zone-course-mission-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getInstructionalServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-instructional-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getSeaCommandServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-sea-command-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getShoreCommandServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-shore-command-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getInterOrganizationServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-inter-organization-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getIntelligenceServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-intelligence-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }


        function getHODServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-hod-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getDockyardServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-dockyard-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getSubmarineServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-submarine-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        

        function getDeputationServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-deputation-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        

        function getOutsideServices(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-outside-services?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        

        function GetFamilyPermissionRelationCount(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-family-permission-relation-count?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        

        function GetFamilyPermissions(pNo, relationId) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-family-permissions?pNo=' + pNo + '&relationId=' + relationId;
            return apiHttpService.GET(url);
        }



        function getForeignProjects(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-foreign-projects?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getMissions(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-missions?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getRemark(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-remark?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getPersuasion(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-persuasion?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getCourseFuturePlan(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-course-future-plan?pNo=' + pNo;
            return apiHttpService.GET(url);
        }
        function getTransferFuturePlan(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-transfer-future-plan?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getNotifications(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-notifications?pNo=' + pNo;
            return apiHttpService.GET(url);
        }


        function getCurrentStatus(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-current-status?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getISSB(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-issb?pNo=' + pNo;
            return apiHttpService.GET(url);
        }



        function getLeaveInfo(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-leave-info?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

        function getAdminAuthorityService(pNo) {
            var url = dataConstants.CURRENT_STATUS_URL + 'get-admin-authority-service?pNo=' + pNo;
            return apiHttpService.GET(url);
        }

    }
})();

File: Infinity.Bnois/Infinity.Bnois.ApplicationService/Implementation/CurrentStatusService.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Threading.Tasks;
using Infinity.Bnois.ApplicationService.Interface;
using Infinity.Bnois.Data;

namespace Infinity.Bnois.ApplicationService.Implementation
{
	public class CurrentStatusService : ICurrentStatusService
	{
		private readonly IBnoisRepository<Employee> employeeRepository;
		private readonly IBnoisRepository<EmployeeGeneral> employeeGeneralRepository;
		private readonly IBnoisRepository<HeirType> heirTypeRepository;
		private readonly IBnoisRepository<Zone> zoneRepository;
		public CurrentStatusService(IBnoisRepository<Employee> employeeRepository, IBnoisRepository<HeirType> heirTypeRepository, IBnoisRepository<Zone> zoneRepository, IBnoisRepository<EmployeeGeneral> employeeGeneralRepository)
		{
			this.employeeRepository = employeeRepository;
			this.heirTypeRepository = heirTypeRepository;
			this.zoneRepository = zoneRepository;
			this.employeeGeneralRepository = employeeGeneralRepository;
		}

		public object GetGeneralInformation(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAnnx1OfficerGeneralReport] '{0}' ", pNo));
			return dataTable.ToJson().FirstOrDefault();
		}

		public List<object> GetCivilAcademicQualification(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetPersonalityBriefByCivilEducationReport] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}
		public List<object> GetSecurityClearance(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetOfficerSecurityClearance] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}


		public List<object> GetCourseAttended(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAnnx6CourseReport] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetForeignCourseAttended(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAnnx6CourseForeignReport] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}


        public object GetForeignCourseVisitGrandTotal(string pNo)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetForeignCourseVisitGrandTotal] '{0}' ", pNo));
            return dataTable.ToJson().FirstOrDefault();
        }
        public List<object> GetExamTestResult(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetExamTestResult] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}
        public List<object> GetUnmDeferment(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetUnmDeferment] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}


        public List<object> GetCareerForecast(string pNo)
		{
			Employee e = employeeRepository.FindOne(x=> x.PNo == pNo);
			EmployeeGeneral eg = employeeGeneralRepository.FindOne(x => x.EmployeeId == e.EmployeeId);
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCareerForecast] '{0}',{1} ", pNo, eg.BranchId));

			return dataTable.ToJson().ToList();
		}
		
		public List<object> GetCarLoanInfo(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCarLoanInfo] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetPunishmentDiscipline(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetPunishmentDiscipline] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetCommendationAppreciation(string pNo,int type)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCommendationAppreciation] '{0}','{1}' ", pNo,type));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetAward(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAward] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetMedal(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetMedal] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetPublication(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetPublication] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}


        public List<object> GetCleanService(string pNo)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCSA] '{0}' ", pNo));

            return dataTable.ToJson().ToList();
        }



        public List<object> GetChildren(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetChildren] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetSibling(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSibling] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}

		public List<object> GetNextOfKin(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetNextOfKin] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}


		private List<object> GetHeir(string pNo, int heirTypeId)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetHeir] '{0}',{1} ", pNo, heirTypeId));

			return dataTable.ToJson().ToList();
		}

		public dynamic GetHeirInfo(string pno)
		{
			var heirTypes = heirTypeRepository.Where(x => x.IsActive).ToList();
            Dictionary<string, dynamic> keyValues = new Dictionary<string, dynamic>();
			foreach (var heirType in heirTypes)
			{
				var vaHeirs = GetHeir(pno, heirType.HeirTypeId);

			    if (vaHeirs.Count >0)
			    {
			        keyValues.Add(heirType.Name, vaHeirs);
                }
				
			}
			return keyValues;
		}

	    public List<object> GetOprGrading(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetOprGrading] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }
	    public List<object> GetForeignVisit(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetForeignVisit] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }
	    public List<object> GetParentInfo(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetParentsInfo] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }


	    public List<object> GetSpouseInfo(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSpouseInfo] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetTransferHistory(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [SpGetTransferHistory] '{0}'", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetPromotionHistory(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [SpGetPromotionHistory] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    private List<object> GetAdditionalSeaServicesByType(string pNo, int type)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAdditionalSeaServices] '{0}',{1} ", pNo, type));

	        return dataTable.ToJson().ToList();
	    }

	    public object GetAdditionalSeaServicesGrandTotal(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAdditionalSeaServicesGrandTotal] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }


	    public dynamic GetAdditionalSeaServices(string pno)
	    {

	        Dictionary<string, dynamic> keyValues = new Dictionary<string, dynamic>();
	        foreach (var type in Enum.GetValues((typeof(ShipType))))
	        {
	            var results = GetAdditionalSeaServicesByType(pno, (int)type);

	            var shipType = String.Empty;
	            if (results.Count > 0)
	            {
	                if ((int)type == (int)ShipType.Small)
	                {
	                    shipType = "Small";

	                }
	                else if ((int)type == (int)ShipType.Medium)
	                {
	                    shipType = "Medium";
	                }
	                else
	                {
	                    shipType = "Large";
	                }


	                keyValues.Add(shipType, results);
	            }

	        }
	        return keyValues;
	    }


	    private List<object> GetSeaServicesByType(string pNo, int type)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSeaServices] '{0}',{1} ", pNo, type));

	        return dataTable.ToJson().ToList();
	    }

	    public object GetSeaServicesGrandTotal(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format(@"declare @GrandTotalDuration varchar(100)
exec [spGetSeaServicesGrandTotal] '{0}',@GrandTotalDuration output
select isnull(@GrandTotalDuration,'') as GrandTotalDuration", pNo));

	        return dataTable.ToJson().ToList();
	    }


	    public dynamic GetSeaServices(string pno)
	    {

	        Dictionary<string, dynamic> keyValues = new Dictionary<string, dynamic>();
	        foreach (var type in Enum.GetValues((typeof(ShipType))))
	        {
	            var results = GetSeaServicesByType(pno, (int)type);

	            var shipType = String.Empty;
	            if (results.Count > 0)
	            {
	                if ((int)type == (int)ShipType.Small)
	                {
	                    shipType = "Small";

	                }
	                else if ((int)type == (int)ShipType.Medium)
	                {
	                    shipType = "Medium";
	                }
	                else
	                {
	                    shipType = "Large";
	                }


	                keyValues.Add(shipType, results);
	            }

	        }
	        return keyValues;
	    }


        public List<object> GetInstructionalServices(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetInstructionalServices] '{0}' ", pNo));

            return dataTable.ToJson().ToList();
		}


		private List<object> GetSeaCommandServicesByType(string pNo,int type)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSeaCommandServices] '{0}',{1} ", pNo,type));

			return dataTable.ToJson().ToList();
		}

	    public object GetSeaCommandServicesGrandTotal(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSeaCommandServicesGrandTotal] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }


        public dynamic GetSeaCommandServices(string pno)
	    {
	       
	        Dictionary<string, dynamic> keyValues = new Dictionary<string, dynamic>();
	        foreach (var type in Enum.GetValues((typeof(ShipType))))
	        {
	            var results = GetSeaCommandServicesByType(pno, (int)type);

	            var shipType = String.Empty;
	            if (results.Count > 0)
	            {
	                if ((int) type == (int)ShipType.Small)
	                {
	                    shipType = "Small";

	                }else if ((int) type == (int) ShipType.Medium)
	                {
	                    shipType = "Medium";
                    }
	                else
	                {
	                    shipType = "Large";
                    }


	                keyValues.Add(shipType, results);
	            }

	        }
	        return keyValues;
	    }



	    private List<object> GetZoneServicesByType(string pNo, int type)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetZoneServices] '{0}',{1} ", pNo, type));

	        return dataTable.ToJson().ToList();
	    }

	    public object GetZoneServicesGrandTotal(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetZoneServicesGrandTotal] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }


	    public dynamic GetZoneServices(string pno)
	    {
	        var zones = zoneRepository.Where(x => x.IsActive).ToList();
            Dictionary<string, dynamic> keyValues = new Dictionary<string, dynamic>();
	        foreach (var item in zones)
	        {
	            var results = GetZoneServicesByType(pno, item.ZoneId);
	         
	            if (results.Count > 0)
	            {
	              

	                keyValues.Add(item.Name, results);
	            }

	        }
	        return keyValues;
	    }


		

	    public dynamic GetZoneCourseMissionServices(string pno)
	    {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetZoneCourseMissionServices] '{0}' ", pno));

            return dataTable.ToJson().ToList();
        }
		
	    public List<object> GetMscEducationQualification(string pno)
	    {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetMscEducationQualification] '{0}' ", pno));

            return dataTable.ToJson().ToList();
        }



        public List<object> GetInterOrganizationServices(string pNo)	
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetInterOrganizationServices] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}
		public List<object> GetIntelligenceServices(string pNo)
		{
			DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetIntelligenceServices] '{0}' ", pNo));

			return dataTable.ToJson().ToList();
		}


	    public List<object> GetMissions(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetMissions] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }


	    public List<object> GetForeignProjects(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetForeignProjects] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }


	    public List<object> GetHODServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetHODServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetDockyardServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDockyardServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetNsdServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetNsdServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetBsdServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetBsdServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetBsoServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetBsoServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetSubmarineServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetSubmarineServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }
		
		

	    public List<object> GetDeputationServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetDeputationServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }
	    public List<object> GetShoreCommandServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetShoreCommandServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetOutsideServices(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetOutsideServices] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }
		

	    public List<object> GetFamilyPermissionRelationCount(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetFamilyPermissionRelationCount] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }
		

	    public List<object> GetFamilyPermissions(string pNo, int relationId)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetFamilyPermissions] '{0}',{1} ", pNo, relationId));

	        return dataTable.ToJson().ToList();
	    }



        public List<object> GetNotifications( string userid, string pNo)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetNotification] '{0}','{1}' ", userid, pNo));

            return dataTable.ToJson().ToList();
        }


        public object GetCurrentStatus(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCurrentStatus] '{0}' ", pNo));
	        return dataTable.ToJson().FirstOrDefault();
	    }


	    public object GetISSB(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetISSB] '{0}' ", pNo));
	        return dataTable.ToJson().FirstOrDefault();
	    }



        public object GetBatchPosition(string pNo)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetBatchPosition] '{0}' ", pNo));

            return dataTable.ToJson().FirstOrDefault();
        }

        public object GetLeaveInfo(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetLeaveInfo] '{0}' ", pNo));
	        return dataTable.ToJson().FirstOrDefault();
	    }


        public List<object> GetTemporaryTransferHistory(int transferId)
	    {

            
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [SpGetTemporaryTransferHistory] {0}", transferId));


	        return dataTable.ToJson().ToList();
        }

	    public List<object> GetPersuasion(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetPersuasion] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
        }

	    public List<object> GetRemark(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetRemark] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
	    }

	    public List<object> GetCourseFuturePlan(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCourseFuturePlan] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
        }

	    public List<object> GetTransferFuturePlan(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetTransferFuturePlan] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
        }
	    public List<object> GetAdminAuthorityService(string pNo)
	    {
	        DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetAdminAuthorityService] '{0}' ", pNo));

	        return dataTable.ToJson().ToList();
        }

        public List<object> GetCostGuardHistory(string pNo)
        {
            DataTable dataTable = employeeRepository.ExecWithSqlQuery(String.Format("exec [spGetCostGuardHistory] '{0}'", pNo));

            return dataTable.ToJson().ToList();
        }
    }
}